### PR TITLE
6 get api articles article id comments

### PR DESCRIPTION
**Pull request by GNicholson95**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -76,3 +86,57 @@ describe('GET/api/articles/:article_id', () => {
...
 });
 
+describe('GET/api/articles/:article_id/comments', () => {
+  it('GET:200 and should respond with an an array of comments for the given article_id', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        console.log(comments);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: This test is very thorough! 🏆  Don't forget to clear out old console logs from when you were building the test 🧹

File: __tests__/app.test.js [hunk 2]
```diff
@@ -76,3 +86,57 @@ describe('GET/api/articles/:article_id', () => {
...
+    });
+    it('should return the comments sorted by date in descending order.', () => {
+      return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response) => {
+          const { comments } = response.body;
+          expect(comments).toBeSortedBy('created_at',{
+            descending: true,
+          });
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:184]: Brilliant

File: __tests__/app.test.js [hunk 3]
```diff
@@ -76,3 +86,57 @@ describe('GET/api/articles/:article_id', () => {
...
+    .then((response) => {
+      expect(response.body.msg).toBe('Bad request')
+  })
+  });
+  it('should throw 404 error if given id doesnt exist', () => {
+    return request(app)
+    .get('/api/articles/1000/comments')
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Path not found')
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: I would check your message here as _"Path not found"_ could be misleading, the error is that the resource can't be found

File: app.js [hunk 4]
```diff
@@ -9,25 +9,25 @@ app.get('/api', getEndpoints);
...
-//     next(error);
-// });
+app.get('/api/articles/:article_id/comments', getCommentsByArticleId);
+
+app.all("/*", (req, res, next) => {
+  res.status(404).send({ msg: "Path not found" });
+});
 
 app.use((err, req, res, next) => {
     console.log(err);
```
Comment (Pipe-Smoking-Rabbit) [inline app.js]: I would recommend moving this console log on line `19` to be inside the _else_ block on line `26 - 28` 
The reason I say this is that currently the log will fire every time you get an error, which will fill your terminal with logs when running the test suite. You only really want that log to fire when it's an unhandled or unexpected error, so it's right at home in the `500` section

File: controllers/controller.js [hunk 5]
```diff
@@ -22,3 +20,10 @@ exports.getArticleById = (req, res, next) => {
 	.catch(next)
   };
 
+
+exports.getCommentsByArticleId  = (req, res, next) => {
+	const { article_id } = req.params;
+	selectCommentsByArticleId(article_id).then((comments) => {
+	  res.status(200).send({ comments });
+	}).catch(next)
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/controller.js:29]: lovely shorthand syntax for the `catch` 👌

File: models/model.js [hunk 6]
```diff
@@ -10,12 +9,22 @@ exports.fetchTopics = () => {
...
 			return Promise.reject({ status: 404, msg: 'Resource not found' });
 		}
 		return result.rows[0];
 	  })
   };
 
+  exports.selectCommentsByArticleId = (Article_id) => {
+	return db
+	  .query('SELECT * FROM comments WHERE comments.Article_id = $1 ORDER BY created_at DESC;', [
+		Article_id
```
Comment (Pipe-Smoking-Rabbit) [inline models/model.js]: be careful with the casing, using a capital first letter in javascript denotes a _class constructor_ , so I'd change `Article_id` to be => `article_id`

File: models/model.js [hunk 7]
```diff
@@ -10,12 +9,22 @@ exports.fetchTopics = () => {
...
+	  ])
+	  .then((result) => {
+		if (result.rows.length === 0) {
+			return Promise.reject({ status: 404, msg: 'Resource not found' });
+		}
```
Comment (Pipe-Smoking-Rabbit) [inline models/model.js]: Do you want this logic to be here... what about when an article has no comments ? 🤔

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Really good work on this ticket! The code is clean and easy to read 👍 

Unfortunately I must ask you to make a change, when an article _exists_ but has no comments, that would be a `200` and respond with an empty array. Currently there is no test for that and the model is throwing a `404`. I have also left comments below in the code

Make that change and then submit another PR for review

File: __tests__/app.test.js (hunk 1 above)
Comment (GNicholson95) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 3 above)
Comment (GNicholson95) [reply __tests__/app.test.js]: done

File: app.js (hunk 4 above)
Comment (GNicholson95) [reply app.js]: done

File: models/model.js (hunk 6 above)
Comment (GNicholson95) [reply models/model.js]: done

File: __tests__/app.test.js [hunk 8]
```diff
@@ -163,7 +142,72 @@ describe('GET/api/articles', () => {
...
+  it('should throw 404 error if given id doesnt exist', () => {
+    return request(app)
+    .get('/api/articles/1000/comments')
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Resource not found')
+  })
+  });
+
+  it('should check exists but has no comments', () => {
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:204]: Looks good👍

**Review by Beth-McCormack — APPROVED** [review]

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**